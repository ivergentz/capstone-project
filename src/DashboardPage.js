import React from 'react'
import styled from 'styled-components/macro'
import { ReactComponent as LogoSvg } from './image/riide.svg'
// import Rides from './RidesList'
// import Bookmarks from './Bookmark'

export default function DashboardPage(isBookmarked) {
  localStorage.getItem('isBookmarked', isBookmarked)
  return (
    <>
      <Header>Deine Fahrten</Header>
      <LogoStyling>
        <LogoSvg />
      </LogoStyling>
      <NoRide>Noch keine Fahrt hinzugef√ºgt</NoRide>
      {/* <Rides isBookmarked={isBookmarked} />
      {console.log(isBookmarked)} */}
    </>
  )
}

const LogoStyling = styled.div`
  display: flex;
  margin: 0 auto;
  margin-top: 40%;
  width: 80vw;
`

const Header = styled.h1`
  display: flex;
  color: var(--button-nav);
  font-weight: 100;
  align-content: center;
  justify-content: center;
  padding: 20px;
  background: var(--back-light);
  border-radius: 0 0 25px 25px;
  box-shadow: 1px 2px 2px 1px rgba(20, 20, 20, 0.2);
  position: fixed;
  width: 100%;
`
const NoRide = styled.p`
  font-size: 0.8em;
  display: grid;
  text-align: center;
  margin-top: 25%;
  color: white;
`
