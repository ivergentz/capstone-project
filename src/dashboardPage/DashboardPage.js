import React, { useEffect, useState } from 'react'
import Rides from './RidesList'
import axios from 'axios'
import styled from 'styled-components/macro'

export default function DashboardPage() {
  const [rides, setRides] = useState([])
  useEffect(() => {
    axios
      .get('http://localhost:5000/rides')
      .then((response) => {
        setRides(response.data)
      })
      .catch(function (error) {
        console.log(error.response.data)
      })
  }, [])

  return (
    <>
      <Header>Deine Fahrten</Header>
      <Rides rides={rides} />
    </>
  )
}

const Header = styled.h1`
  display: flex;
  color: var(--button-nav);
  font-weight: 100;
  align-content: center;
  justify-content: center;
  padding: 20px;
  background: var(--back-light);
  border-radius: 0 0 25px 25px;
  box-shadow: 1px 2px 2px 1px rgba(0, 51, 0, 0.2);
  position: fixed;
  width: 100%;
`
