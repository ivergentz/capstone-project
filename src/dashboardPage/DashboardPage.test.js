import React from 'react'
import renderer from 'react-test-renderer'
import DashboardPage from './DashboardPage'

it('renders correctly', () => {
  const tree = renderer
    .create(<DashboardPage />)
  
  expect(tree).toMatchSnapshot();
})
