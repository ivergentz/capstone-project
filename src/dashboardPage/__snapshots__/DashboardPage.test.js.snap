// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="DashboardPage__Header-sc-17821ts-0 IdgAV"
  >
    Deine Fahrten
  </h1>,
  <p>
    Noch keine Fahrt hinzugef√ºgt
  </p>,
]
`;
